pandas/tests/test_interval.py

```diff
@@ -0,0 +1,527 @@
+        self.assertTrue(Interval(0, 1) >= Interval(0, 1))
+
+    def test_hash(self):
+        # should not raise
+        hash(self.interval)
+
+    # def test_math(self):
+    #     expected = Interval(1, 2)
+    #     actual = self.interval + 1
+    #     self.assertEqual(expected, actual)
+
+
+class TestIntervalTree(tm.TestCase):
+    def setUp(self):
+        self.tree = IntervalTree(np.arange(5), np.arange(5) + 2)
+
+    def test_get_loc(self):
+        self.assert_numpy_array_equal(self.tree.get_loc(1), [0])
+        self.assert_numpy_array_equal(np.sort(self.tree.get_loc(2)), [0, 1])
+        with self.assertRaises(KeyError):
+            self.tree.get_loc(-1)
+
+    def test_get_indexer(self):
+        self.assert_numpy_array_equal(
+            self.tree.get_indexer(np.array([1.0, 5.5, 6.5])), [0, 4, -1])
+        with self.assertRaises(KeyError):
+            self.tree.get_indexer(np.array([3.0]))
+
+    def test_get_indexer_non_unique(self):
+        indexer, missing = self.tree.get_indexer_non_unique(
+            np.array([1.0, 2.0, 6.5]))
+        self.assert_numpy_array_equal(indexer[:1], [0])
+        self.assert_numpy_array_equal(np.sort(indexer[1:3]), [0, 1])
+        self.assert_numpy_array_equal(np.sort(indexer[3:]), [-1])
+        self.assert_numpy_array_equal(missing, [2])
+
+    def test_duplicates(self):
+        tree = IntervalTree([0, 0, 0], [1, 1, 1])
+        self.assert_numpy_array_equal(np.sort(tree.get_loc(0.5)), [0, 1, 2])
+
+        with self.assertRaises(KeyError):
+            tree.get_indexer(np.array([0.5]))
+
+        indexer, missing = tree.get_indexer_non_unique(np.array([0.5]))
+        self.assert_numpy_array_equal(np.sort(indexer), [0, 1, 2])
+        self.assert_numpy_array_equal(missing, [])
+
+    def test_get_loc_closed(self):
+        for closed in ['left', 'right', 'both', 'neither']:
+            tree = IntervalTree([0], [1], closed=closed)
+            for p, errors in [(0, tree.open_left),
+                              (1, tree.open_right)]:
+                if errors:
+                    with self.assertRaises(KeyError):
+                        tree.get_loc(p)
+                else:
+                    self.assert_numpy_array_equal(tree.get_loc(p),
+                                                  np.array([0]))
+
+    def test_get_indexer_closed(self):
+        x = np.arange(1000)
+        found = x
+        not_found = -np.ones(1000)
+        for closed in ['left', 'right', 'both', 'neither']:
+            tree = IntervalTree(x, x + 0.5, closed=closed)
+            self.assert_numpy_array_equal(found, tree.get_indexer(x + 0.25))
+
+            expected = found if tree.closed_left else not_found
+            self.assert_numpy_array_equal(expected, tree.get_indexer(x + 0.0))
+
+            expected = found if tree.closed_right else not_found
+            self.assert_numpy_array_equal(expected, tree.get_indexer(x + 0.5))
+
+
+class TestIntervalIndex(tm.TestCase):
```

Comment:
ok, but why don't you add a sub-class in ``test_index.py`` and put a note there. AND inherit this from ``Base`` as well. The reason these are all in the same file is to push as much testing into ``Base`` so sub-classes are as consistent as possible.